### PR TITLE
Make jsoncpp:fPIC=False work on 1.9.3

### PR DESCRIPTION
**Pull request by mathbunnyru**
Specify library name and version:  **jsoncpp/1.9.3**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by mathbunnyru** [comment]
In the latest version of jsoncpp, they do set target properties in a different way:

{code}
set_target_properties( jsoncpp_lib PROPERTIES
    OUTPUT_NAME jsoncpp
    VERSION ${JSONCPP_VERSION}
    SOVERSION ${JSONCPP_SOVERSION}
    POSITION_INDEPENDENT_CODE ON
)
{code}

My change should make it work with old versions and the latest one as well.

`conan create all jsoncpp/1.9.3@ -o jsoncpp:fPIC=False` - I tried this with both 1.9.2 and 1.9.3 and it seems to work.

File: recipes/jsoncpp/all/conanfile.py [hunk 1]
```diff
@@ -37,8 +37,8 @@ def _patch_sources(self):
 
         if self.settings.os != "Windows" and not self.options.shared and not self.options.fPIC:
             tools.replace_in_file(os.path.join(self._source_subfolder, "src", "lib_json", "CMakeLists.txt"),
-                                  "set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)",
-                                  "set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE OFF)")
+                                  "POSITION_INDEPENDENT_CODE ON",
+                                  "POSITION_INDEPENDENT_CODE OFF")
```
Comment (prince-chrismc) [inline recipes/jsoncpp/all/conanfile.py:41]: Perhaps this could be a patch (to remove it) for the two versions in questions in the `conandata.yml`

Comment (mathbunnyru) [reply recipes/jsoncpp/all/conanfile.py:41]: If it's a better approach, than I can do it, I suppose.

Comment (prince-chrismc) [reply recipes/jsoncpp/all/conanfile.py:41]: Let the CCI team answer that :stuck_out_tongue_closed_eyes:

Comment (uilianries) [reply recipes/jsoncpp/all/conanfile.py:41]: YOu can literally remove it from the cmake file, the conan cmake wrapper, using conan_basic_setup sets it globally. e.g. replace("set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)", "")

This recipe has many small patches, but them depends on the version, or an option, you can create patches of course, but requires more effort as you need to add new conditions in cmake. For this one, you keep `tools.replace_in_file`

Comment (uilianries) [inline recipes/jsoncpp/all/conanfile.py:41]:
```suggestion
                                  "set_target_properties( jsoncpp_lib PROPERTIES POSITION_INDEPENDENT_CODE ON)",
                                  "")
```

Comment (mathbunnyru) [reply recipes/jsoncpp/all/conanfile.py]: Unfortunately, this won't work with 1.9.3, because there is no such a lign.

Comment (mathbunnyru) [reply recipes/jsoncpp/all/conanfile.py]: I made a replace, which basically deletes this options.
It depends on the version of JsonCpp, but there is no if for OS or something else

Comment (mathbunnyru) [reply recipes/jsoncpp/all/conanfile.py:41]: Unfortunately, doesn't seem to work. So I falled back to my original version.

**Review by SSE4 — APPROVED** [review]

**Review by danimtb — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**